extractor/docker/docker.go

```diff
@@ -60,19 +60,29 @@ type layer struct {
 	Content io.ReadCloser
 }
 
-type opqDirs []string
-
 type DockerExtractor struct {
+	Client *client.Client
+	Cache  cache.Cache
 	Option types.DockerOption
 }
 
-func NewDockerExtractor(option types.DockerOption) DockerExtractor {
+func NewDockerExtractor(option types.DockerOption) (extractor.Extractor, error) {
```

Comment:
Returning `interface` is cumbersome for the receiver. If they need interface, they can always cast it. Casting the other way (to a specific struct) is a bug-friendly practice.

Reply:
I agree. Accepting structs and returning interfaces is a good practice.